Tools/TestWebKitAPI/Tests/WebCore/TextBoundaries.cpp

```diff
@@ -37,7 +37,22 @@ TEST(TextBoundariesTest, FindWordBoundaryEmpty)
 
     WebCore::findWordBoundary(stringView, 0, &start, &end);
 
-    ASSERT(!start);
-    ASSERT(!end);
+    EXPECT_EQ(start, 0);
+    EXPECT_EQ(end, 0);
+}
+
+TEST(TextBoundariesTest, FindWordBoundaryPositionTooBig)
+{
+    WTF::String aString("Hello World"_s);
+    WTF::StringView stringView(aString);
+    int start = -1;
+    int end = -1;
+
+    WebCore::findWordBoundary(stringView, aString.length(), &start, &end);
+
+#if !USE(APPKIT)
+    EXPECT_EQ(start, static_cast<int>(aString.length() - 1));
+    EXPECT_EQ(end, static_cast<int>(aString.length() - 1));
```

Comment:
Can we provide expected results for the `USE(APPKIT)` code path, or do they vary on each run?

Reply:
I'm not entirely sure, but the other implementations seem to use different libary or API calls so I don't think we can guarantee the output here.

https://searchfox.org/wubkat/source/Source/WebCore/platform/text/mac/TextBoundaries.mm#181
https://searchfox.org/wubkat/source/Source/WebCore/platform/text/TextBoundaries.cpp#93